### PR TITLE
Define and adopt standard IP ranges for FAST networking

### LINKED ISSUE #1644: stages: gke-multitenant complains with `IP_SPACE_EXHAUSTED` error
Deploying the [3-gke-multitenant](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/tree/master/fast/stages/3-gke-multitenant/dev) stage with this configuration:

```ini
clusters = {
  cluster-0 = {
    location = "us-east1"
    vpc_config = {
      subnetwork = "https://www.googleapis.com/compute/v1/projects/XXX-dev-net-spoke-0/regions/us-east1/subnetworks/dev-default-ue1"
    }
  }
}

nodepools = {
  cluster-0 = {
    nodepool-0 = {
      max_pods_per_node = 10
      node_config = {
        disk_type    = "pd-ssd"
        machine_type = "c2-standard-16"
        spot         = true
      }
    }
  }
}

fleet_features = null

fleet_workload_identity = false
```

This is using the default values already available in the [2-networking-d-separate-envs](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/fast/stages/2-networking-d-separate-envs/data/subnets/dev/dev-dataplatform-ew1.yaml) stage:

```ini
region: us-east1
ip_cidr_range: 10.127.48.0/24
description: Default subnet for dev
secondary_ip_ranges:
   pods: 100.64.0.0/24
   services: 100.64.1.0/24
```

### PR DESCRIPTION
**Pull request by juliocc**
This PR documents and adopts a consistent IP address plan for FAST networking stages

Fixes #1644

---
**Checklist**

I applicable, I acknowledge that I have:
- [x] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass

### CODE REVIEW

File: fast/stages/2-networking-c-nva/README.md [hunk 1]
```diff
@@ -121,13 +121,13 @@ This is an options summary:
...
 
-The /16 block is evenly split in eight, smaller /19 blocks, assigned to different areas of the GCP network: *landing untrusted europe-west1*, *landing untrusted europe-west4*, *landing trusted europe-west1*, *landing untrusted europe-west4*, *development europe-west1*, *development europe-west4*, *production europe-west1*, *production europe-west4*.
+The /11 block is evenly split in eight, smaller /16 blocks, assigned to different areas of the GCP network: *landing untrusted europe-west1*, *landing untrusted europe-west4*, *landing trusted europe-west1*, *landing untrusted europe-west4*, *development europe-west1*, *development europe-west4*, *production europe-west1*, *production europe-west4*.
 
 The first /24 range in every area is allocated for a default subnet, which can be removed or modified as needed.
 
-Spoke VPCs also define and reserve three "special" CIDR ranges, derived from the respective /19, dedicated to
+Spoke VPCs also define and reserve three "special" CIDR ranges, derived from their respective /16, dedicated to
```
Comment (ludoo) [inline fast/stages/2-networking-c-nva/README.md:130]: I don't think we book PSA ranges and proxy subnets anymore.

**Review by ludoo — APPROVED** [review]

File: fast/stages/2-networking-c-nva/README.md [hunk 2]
```diff
@@ -147,24 +147,24 @@ This is a summary of the subnets allocated by default in this setup:
...
+| prod-default-ew1 | Prod spoke subnet - europe-west1 | 10.72.0.0/24 |
+| prod-default-ew1 (PSA MySQL) | PSA subnet for MySQL in prod spoke - europe-west1 | 10.72.253.0/24 |
+| prod-default-ew1 (PSA SQL Server) | PSA subnet for Postgres in prod spoke - europe-west1 | 10.72.254.0/24 |
+| prod-default-ew1 (L7 ILB) | L7 ILB subnet for prod spoke - europe-west1 | 10.72.255.0/24 |
+| prod-default-ew4 | Prod spoke subnet - europe-west4 | 10.88.0.0/24 |
+| prod-default-ew4 (PSA MySQL) | PSA subnet for MySQL in prod spoke - europe-west4 | 10.88.253.0/24 |
+| prod-default-ew4 (PSA SQL Server) | PSA subnet for Postgres in prod spoke - europe-west4 | 10.88.254.0/24 |
+| prod-default-ew4 (L7 ILB) | L7 ILB subnet for prod spoke - europe-west4 | 10.88.255.0/24 |
+
+These subnets are advertised to on-premises as a whole /11 range (10.64.0.0/11).
```
[the remark this answers is not in the document]
Comment (sruffilli) [reply fast/stages/2-networking-c-nva/README.md:167]: We don't anymore - we used to provide a default value for `var.vpn_onprem_configs` which used to advertise named ranges (including `gcp_all`) to onprem. We can remove that.

**Review by sruffilli — APPROVED** [review]
LGTM in principle - only nit is that the allocated IP space is seldom realistic on real world scenarios. 

Item for a future TODO: It would be handy to have a short document supporting users in designing their IP space, highlighting what's beyond VMs (especially PSA ranges for the different managed services, GKE cookiemonster).

File: fast/docs/1-network-ranges.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- Support for 2 regions
+- Leave enough space to easily grow either the number of environments or regions
+- Allocate large blocks from the CG-NAT range to use as secondary ranges, primarily for GKE pods and services.
+
+The following table summarizes the agreed IP plan:
+
+| |Aggregate| landing | dev | prod|
+|---|---:|---:|---:|---:|
+|Region 1, primary ranges|10.64.0.0/12|10.64.0.0/16<br>Trusted: 10.64.0.0/17<br>Untrusted: 10.64.128.0/17|10.68.0.0/16|10.72.0.0/16|
+|Region 2, primary ranges|10.80.0.0/12|10.80.0.0/16<br>Trusted: 10.80.0.0/17<br>Untrusted: 10.80.128.0/17|10.68.0.0/16|10.72.0.0/16|10.84.0.0/16|10.88.0.0/16|
```
[the remark this answers is not in the document]
Comment (juliocc) [reply fast/docs/1-network-ranges.md]: fixed

File: fast/stages/2-networking-c-nva/README.md (hunk 1 above)
Comment (juliocc) [reply fast/stages/2-networking-c-nva/README.md:130]: I'll open a new issue to update these readmes. There's a few things that are no longer true.

**Pull request merged**